pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -311,6 +320,88 @@ func createIngress(isvc *v1beta1.InferenceService, config *v1beta1.IngressConfig
 		hosts = append(hosts, serviceHost)
 		gateways = append(gateways, config.IngressGateway)
 	}
+	if config.UrlTemplate != "" {
+		url, err := GenerateUrl(isvc.Name, isvc.Namespace, config)
+		if err != nil {
+			log.Error(err, "Failed to generate URL from urlTemplate")
+			return nil
+		}
+		url.Path = strings.TrimSuffix(url.Path, "/") // remove trailing "/" if present
+		if url.Path != "" {
+			// In this case, we have a path-based URL so we add a path-based rule
+			httpRoutes = append(httpRoutes, &istiov1alpha3.HTTPRoute{
+				Match: []*istiov1alpha3.HTTPMatchRequest{
+					{
+						Uri: &istiov1alpha3.StringMatch{
+							MatchType: &istiov1alpha3.StringMatch_Prefix{
+								Prefix: url.Path + "/",
+							},
+						},
+						Authority: &istiov1alpha3.StringMatch{
+							MatchType: &istiov1alpha3.StringMatch_Regex{
+								Regex: constants.HostRegExp(url.Host),
+							},
+						},
+						Gateways: []string{config.IngressGateway},
+					},
+					{
+						Uri: &istiov1alpha3.StringMatch{
+							MatchType: &istiov1alpha3.StringMatch_Exact{
+								Exact: url.Path,
+							},
+						},
+						Authority: &istiov1alpha3.StringMatch{
+							MatchType: &istiov1alpha3.StringMatch_Regex{
+								Regex: constants.HostRegExp(url.Host),
+							},
+						},
+						Gateways: []string{config.IngressGateway},
+					},
+				},
+				Rewrite: &istiov1alpha3.HTTPRewrite{
+					Uri: "/",
```

Comment:
I am not sure how this is going to work, for example if the `UrlTemplate` is `"https://{{ .IngressDomain }}/{{ .Namespace }}/{{ .Name }}` and the user is calling with the following url `http://istio-ingressgateway.istio-system/user/isvc/v2/models/cifar/infer`, `url.path` would return `user/isvc` and rewrite to `/`.